components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/AnalyticsUtils.java

```diff
@@ -0,0 +1,146 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.gateway.handlers.analytics;
+
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.rest.RESTConstants;
+import org.wso2.carbon.apimgt.gateway.APIMgtGatewayConstants;
+import org.wso2.carbon.apimgt.usage.publisher.dto.enums.EVENT_TYPE;
+import org.wso2.carbon.apimgt.usage.publisher.dto.enums.FAULT_EVENT_TYPE;
+import org.wso2.carbon.apimgt.gateway.handlers.security.APISecurityUtils;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+
+import java.util.Map;
+
+/**
+ * Implementation of the util methods
+ */
+public class AnalyticsUtils {
+    private static final String ANALYTICS_REQUEST_TYPE = "analytics_request_type";
+    private static final String THROTTLED_OUT_REQUEST = "throttled_out_request";
+
+    public static void markedAsThrottled(MessageContext messageContext) {
+        messageContext.setProperty(ANALYTICS_REQUEST_TYPE, THROTTLED_OUT_REQUEST);
+    }
+
+    public static boolean isRequestThrottledOut(MessageContext messageContext) {
+        if (messageContext.getPropertyKeySet().contains(ANALYTICS_REQUEST_TYPE)) {
+            String type = (String) messageContext.getProperty(ANALYTICS_REQUEST_TYPE);
+            return THROTTLED_OUT_REQUEST.equals(type);
+        } else {
+            return false;
+        }
+    }
+
+    public static boolean isAuthFaultRequest(int errorCode) {
+        return errorCode >= 900900 && errorCode < 901000;
```

Comment:
Can't we read these values from APIThrottleConstants and other constant classes?

Reply:
here we have used a range. Even though we have denied specific codes in constant, we don't have constant for range. We will define a new constant for these two and use it.

Reply:
Fixed